### PR TITLE
Treat hybrid touch laptops as desktop via shared touch-first detection

### PR DESCRIPTION
**Pull request by punk6529**
## Issue
- Windows touch-screen laptops (e.g. Surface) were permanently classified as mobile: any window narrower than 1024px switched the whole app to `SmallScreenLayout`, and even at full width drops lost hover actions, tooltips were suppressed, long-press/bottom-sheet menus appeared, and interactive artwork was tap-gated.
- Root cause: `useDeviceInfo.hasTouchScreen` was raw `maxTouchPoints > 0`, plus ~10 components ran their own `(pointer: coarse)` / `maxTouchPoints` / `ontouchstart` checks. Detection was decentralized, so earlier partial fixes (e.g. `useIsTouchDevice`) never fixed the whole surface.

## Fix
- New `helpers/touch-first.helpers.ts` is the single sanctioned definition: a device is *touch-first* only when touch exists AND there is no fine pointer (`any-pointer: fine`) AND no hover support (`any-hover: hover`). Hybrids (touch + trackpad/mouse) therefore get the desktop experience; phones/tablets keep the touch experience.
- `useIsTouchDevice` rebuilt on `useSyncExternalStore` with live capability-change subscriptions (mouse plug/unplug, convertible posture flips update without reload).
- `useDeviceInfo.hasTouchScreen` now means touch-first (raw capability retained internally only for iPad-desktop-UA detection), which corrects `LayoutWrapper` and ~30 consumers at once.
- All standalone detection sites converted to the shared helpers: sidebar controller, WebSidebar, WaveSettingRow, wave pin, small-leaderboard outcomes, profile statements (4), chat link buttons, dropdown sheet wrapper, quick-vote bar, launch telemetry, leaderboard CSS.
- Phone hardening: `LayoutWrapper` small layout requires `(hasTouchScreen || isMobileDevice)` — a phone with a mouse attached stays on the mobile layout via its UA even though touch-first flips false. Android tablets report `userAgentData.mobile=false`, so tablet+trackpad correctly gets desktop.

## Changes
- New: `helpers/touch-first.helpers.ts`, tests for it, and a `LayoutWrapper` layout-decision test matrix.
- Behavior change for hybrid devices only: desktop layout/affordances at every width. Pure touch devices and the Capacitor app (`isApp` → `MobileLayout`) are unchanged.
- Tests that encoded the old hybrid behavior were deliberately flipped (`useDeviceInfo`, `useDropActionInteractionMode`, `WaveDrop`, `ChatItemHrefButtons`).

## Validation
- `jest`: 96+ affected suites green, including new regression cases (hybrid + narrow window → WebLayout; phone UA + mouse → SmallScreenLayout).
- `typecheck:ci` and default-config eslint on all touched files: clean.
- Playwright against a local dev server: Surface profile (maxTouchPoints=10 + fine pointer + hover + coarse) at 950px and 1600px → desktop layout; plain-desktop control → desktop; phone profile → small-screen layout.
- Not tested: real convertible posture flips (Windows slate mode) — covered by design via media-query change subscriptions; residual risk is a device staying on desktop UI in tablet posture if the OS keeps the touchpad enumerated, which is benign.

## Risk
- Level: Medium
- Why: touches the app-wide layout selector and shared interaction-mode hooks; blast radius is UI-mode selection. Logic is centralized, unit-tested per device profile, and verified in a running app. No data, auth, or API changes.
- Rollback: revert this single commit; no migrations or config involved.

## Review Notes
- The semantic change of `hasTouchScreen` (capability → touch-first) is intentional and documented in the interface JSDoc; all consumers want touch-first semantics.
- `useHasTouchInput` intentionally remains raw capability but is hover-guarded at its only decision site (`useDropActionInteractionMode`).
- `(any-hover: none) and (any-pointer: coarse)` CSS in globals/tailwind was already hybrid-safe and left untouched.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved touch vs. hover handling on hybrid touch laptops and compact viewports, fixing when tooltips, overlays, and action interactions appear.
  * Updated responsive layout behavior so the correct sidebar and wave/UI layouts render for touch-first devices, mobile-web, and native-app scenarios.
  * Refined small-screen and coarse-pointer styling to better match expected touch behavior.
* **Refactor**
  * Standardized touch-device detection across the app for consistent UI behavior.
* **Tests**
  * Expanded coverage for touch-first, hybrid, and breakpoint edge cases to prevent regressions.

### CODE REVIEW

**Comment by punk6529** [comment]
Addressed the review feedback in 762a488e7:

- **Phone with paired mouse / hovering stylus (both Important findings)**: fixed at the source. `isTouchFirstEnvironment` now short-circuits to touch-first for mobile user agents (client hints take precedence, so Android tablets with trackpads still get the desktop experience). Every consumer — `useIsTouchDevice`, `useDeviceInfo.hasTouchScreen`, `WaveSettingRow`, statement tooltips, dropdown sheets — inherits the override, so the standalone call sites no longer need per-site guards. New tests cover phone+mouse (client hints), stylus phone (UA regex fallback), and tablet+trackpad (client hints mobile=false → desktop).
- **BrainLeftSidebarWavePin manual resize detection (nice-to-have)**: converted to the reactive `useIsTouchDevice()` hook.
- **WaveSmallLeaderboardItemOutcomes one-shot SSR snapshot (nice-to-have)**: converted to `useIsTouchDevice()` as well.
- **SonarCloud S7764**: `window.matchMedia` → `globalThis.matchMedia` in the telemetry helper.

Validation: helper/useDeviceInfo/useIsTouchDevice/WavePin/LayoutWrapper suites green (45 tests), debt ratchet passes, eslint clean on all touched files.

**Comment by punk6529** [comment]
**Responsiveness lane finding — flake evidence (rerun-once policy):**

The single blocking finding (`native-mobile` `/the-memes`: `body.capacitor-native` not applied) does not reproduce and is unrelated to this diff:

- `body.capacitor-native` is applied by `components/providers/CapacitorSetup.tsx`, which this PR does not touch; nothing in the diff feeds its Capacitor detection.
- The bot's own run applied the class on 11/12 native routes (including `/the-memes`'s siblings), and the failing capture co-occurred with full-page screenshot timeouts on that image-heavy route — consistent with the timing race the lane itself suggested.
- Local rerun on this branch: the repo's own contract tests asserting `body.capacitor-native` + native runtime signals (`tests/surfaces/native-shell-readonly.spec.ts`) pass on both `capacitor-ios-sim` and `capacitor-android-sim`.
- The same responsiveness run reports `web-desktop` and `web-mobile` — the two surfaces this PR actually changes — clean with 0 failures, and explicitly confirms SmallScreenLayout still activates for `web-mobile`.

Also pushed 4b7780956 adding the tablet-viewport (750–1024px) branch to the LayoutWrapper decision-matrix test per the GLM swarm advisory. Remaining GLM advisory items reviewed: the `useHasTouchInput` mock seam is still the real seam (the hook intentionally consumes raw capability with a hover guard), the touch-only-wide-viewport combination is reachable (large tablets), and the helper takes `touchDetected` as a parameter — no module state to leak.

File: components/waves/small-leaderboard/WaveSmallLeaderboardItemOutcomes.tsx [hunk 1]
```diff
@@ -15,9 +16,7 @@ interface WaveSmallLeaderboardItemOutcomesProps {
 const WaveSmallLeaderboardItemOutcomesContent: React.FC<
   WaveSmallLeaderboardItemOutcomesProps
 > = ({ drop, isMobile: _isMobile = false }) => {
-  const [isTouch] = useState(
-    () => typeof globalThis !== "undefined" && "ontouchstart" in globalThis
-  );
+  const [isTouch] = useState(() => isTouchFirstEnvironment());
```
[the remark this answers is not in the document]
Comment (punk6529) [reply components/waves/small-leaderboard/WaveSmallLeaderboardItemOutcomes.tsx]: Fixed before this comment landed — 762a488e7 converted this component to the reactive `useIsTouchDevice()` hook, which subscribes to capability media-query changes.

File: hooks/useDeviceInfo.ts [hunk 2]
```diff
@@ -1,10 +1,21 @@
 "use client";
 
 import { useState, useCallback, useEffect, useRef } from "react";
+import {
+  hasFinePointerCapability,
+  hasHoverCapability,
+  subscribeToTouchFirstChanges,
+} from "@/helpers/touch-first.helpers";
```
[the remark this answers is not in the document]
Comment (punk6529) [reply hooks/useDeviceInfo.ts:8]: Fixed in 3b359bda5 — `hasTouchInput` now rides the shared `hasTouchCapability()` (touchstart observation still included), so the iPad-UA disambiguation and the touch-first helpers can no longer diverge.

File: helpers/touch-first.helpers.ts [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+    | undefined;
+  if (!nav) {
+    return false;
+  }
+  const uaDataMobile = nav.userAgentData?.mobile;
+  if (typeof uaDataMobile === "boolean") {
+    return uaDataMobile;
+  }
+  return MOBILE_USER_AGENT_REGEX.test(nav.userAgent);
+};
```
[the remark this answers is not in the document]
Comment (punk6529) [reply helpers/touch-first.helpers.ts:98]: Fixed in 3b359bda5 — regex narrowed to `/iPhone|iPod|BlackBerry|IEMobile|Opera Mini|Android.*Mobile/i` and a regression test added for an Android tablet UA without client hints (fine pointer + hover → desktop).

**Comment by punk6529** [comment]
**Deployment bus — staging manifest**

- environment: staging
- staging_branch: `1a-staging`
- staging_source_ref: `fix/touch-first-device-detection` (head 3b359bda5)
- staging merge commit: e346bdec9
- production_target_branch: `main`
- validated_release_set: PR #3075 only (PR #3072 merged to main earlier and is already part of the main base)
- release_equivalence: production candidate = main + #3075 → same patch set as staged
- backend_dependencies: none
- risk: Medium (app-wide UI-mode selection); rollback = revert PR, no migrations
- validation: auto `Staging E2E` workflow (12 packs, desktop+mobile) + release-specific hybrid/phone device-profile checks against staging.6529.io
- validation owner: this session (release captain)
- status: deploying

**Comment by punk6529** [comment]
**Staging validation status — hold + fix-forward**

- Staging deploy e346bdec9: ✅ success (run 28742783662)
- Release-specific device-profile checks against deployed staging: ✅ 5/5 PASS (Surface hybrid 950px/1600px → desktop; plain desktop control → desktop; phone → small-screen; phone+mouse → small-screen), zero page errors
- `Staging E2E` workflow: ❌ 1 failing test — `waves-profile-readonly` `/punk6529/xtdh`: readonly guard blocks the page's two alchemy read-proxy POSTs. **Pre-existing**: identical failure on run 28741346113 (before this PR entered staging, which also had 2 additional failures that now pass). No staging run of this new workflow has ever been green.
- Disposition per deployment-bus: production held; fix-forward via PR #3086 (test-support allowlist for the two read-proxy endpoints). After #3086 merges, main→1a-staging sync + Staging E2E rerun; promotion proceeds once green.

**Comment by punk6529** [comment]
**Staging E2E rerun (62e89f846, run 28744182752) — disposition and promotion decision**

- ✅ xtdh readonly-guard failure: **cured** by PR #3086.
- ✅ Release-specific device profiles: 5/5 PASS on the rerun build (10/10 across both staging builds).
- ❌ Remaining failures: `nextgen-collections-readonly` (horizontal-overflow assertions on NextGen landing/about) and `media-mint-detail-readonly` (title "Mint | The Memes" — no active mint today, expectation assumes a numbered mint).
- **Baseline proof**: the identical packs run against production (`6529.io`, deployed 11:16Z today, pre-release) fail with byte-identical errors — same overflow offenders (`right:1291,width:1223`), same title mismatch — plus additional nextgen failures. These are pre-existing site/test debt, not regressions of this release set; the nextgen failure also appeared on the 12:50Z staging run before this PR entered staging.
- Disposition per deployment-bus: environment/baseline issues documented and handed off as follow-up work; the release-critical surfaces (layout selection, drops, tooltips, home/about/memes smoke, social, messages, delegation, network-open-data, public-content, search) are green on staging. Proceeding with production promotion for the validated release set (main + #3075).

**Comment by punk6529** [comment]
**Deployment bus — closeout: 4.70.0 in production**

- Production: `Web Deploy - PROD` run 28745717051, main @ 7d59e7322 — success
- Staging validation: Staging E2E run 28745223584 fully green (12 packs) on the exact production candidate; device profiles 15/15 across staging builds
- Production validation: device profiles 5/5 on 6529.io; nextgen/media-mint/home packs 17/17 (both previously failing on the pre-release prod baseline)
- main merged back into 1a-staging (9981e5e48)
- Release note: 4.70.0 in 6529 Releases (drop #1165926); deployment overview posted to Follow The Repo
- manifest status: **passed**
